src/content/libraries/async/using-streams.md

```diff
@@ -304,27 +304,62 @@ Stream<T> timeout(
 Stream<S> transform<S>(StreamTransformer<T, S> streamTransformer);
 ```
 
-The final three functions are more special.
-They involve error handling which an **await for** loop
-can't do—the first error reaching the loops will end
-the loop and its subscription on the stream.
-There is no recovering from that.
-The following code shows how to use `handleError()` to remove errors
-from a stream before using it in an **await for** loop.
+The final three functions are more specialized.
+They involve error handling that an **await for** loop
+cannot directly manage; the first error encountered will
+terminate the loop and its stream subscription, with no
+built-in mechanism for recovery.
 
-<?code-excerpt "misc/lib/tutorial/misc.dart (map-log-errors)"?>
-```dart
+The following code demonstrates how to use `handleError()`
+to filter out errors from a stream before it's consumed by
+an **await for** loop.
+
+<?code-excerpt "misc/lib/tutorial/misc.dart (map-log-errors)" plaster="none"?>
+```dart highlightLines=5
 Stream<S> mapLogErrors<S, T>(
   Stream<T> stream,
   S Function(T event) convert,
 ) async* {
   var streamWithoutErrors = stream.handleError((e) => log(e));
+
   await for (final event in streamWithoutErrors) {
     yield convert(event);
   }
 }
 ```
 
+In the previous example, an **await for** loop is never
+returned to if no events are emitted by the stream.
+To avoid this, use the `timeout()` function to create
+a new stream. `timeout()` enables you to set a
+time limit and continue emitting events on the returned
+stream.
+
+The following code modifies the previous example. 
```

Comment:
The entire section had a lot of pre-existing wording issues. I've gone through and cleaned things up. Double-check this last part.